Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -2141,35 +2206,38 @@ def direct_message_handler(integration_context: dict, request_body: dict, conver
 
     allow_external_incidents_creation: bool = demisto.params().get('allow_external_incidents_creation', False)
 
-    lowered_message = message.lower()
-    if lowered_message.find('incident') != -1 and (lowered_message.find('create') != -1
-                                                   or lowered_message.find('open') != -1
-                                                   or lowered_message.find('new') != -1):
-        if user_email:
-            demisto_user = demisto.findUser(email=user_email)
-        else:
-            demisto_user = demisto.findUser(username=username)
+    demisto_user = demisto.findUser(email=user_email) if user_email else demisto.findUser(username=username)
 
-        if not demisto_user and not allow_external_incidents_creation:
-            data = 'You are not allowed to create incidents.'
-        else:
+    lowered_message = message.lower()
+    # the command is to create new incident
+    create_incident = lowered_message.find('incident') != -1 and (lowered_message.find('create') != -1
+                                                                  or lowered_message.find('open') != -1
+                                                                  or lowered_message.find('new') != -1)
+    data = ""
+    if not demisto_user:
+        data = handle_external_user(user_email if user_email else username,
```

Comment:
you can use to `or` operator to simplify this.
```suggestion
        data = handle_external_user(user_email or username,
```